enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java

```diff
@@ -238,45 +240,69 @@ public void execute(@Nonnull EnforcerRuleHelper helper) throws EnforcerRuleExcep
-            throw new EnforcerRuleException(
-                "Artifact " + Artifacts.toCoordinates(artifact) + " is not associated with a file."
-                    + " The linkage checker enforcer rule should be bound to the verify phase.");
-          }
-          Path path = file.toPath();
-          builder.add(path);
-        }
-      }
-      return builder.build();
+      return buildClasspath(resolutionResult);
     } catch (ComponentLookupException e) {
       throw new EnforcerRuleException("Unable to lookup a component " + e.getMessage(), e);
     } catch (DependencyResolutionException e) {
-      Log logger = helper.getLog();
-      for (Throwable cause = e.getCause(); cause != null; cause = cause.getCause()) {
-        if (cause instanceof ArtifactTransferException) {
-          ArtifactTransferException artifactException = (ArtifactTransferException) cause;
-          Artifact artifact = artifactException.getArtifact();
-          DependencyNode dependencyRoot = e.getResult().getDependencyGraph();
-          String pathsToArtifact = findPaths(dependencyRoot, artifact);
+      return buildClasspathFromException(e);
+    }
+  }
+
+  /**
+   * Returns class path built from partial dependency graph of {@code resolutionException}.
+   *
+   * @throws EnforcerRuleException when {@code resolutionException} is invalidated by {@link
+   *     DependencyGraphBuilder#isUnacceptableResolutionExceptionAt(List)}
+   */
+  private ImmutableList<Path> buildClasspathFromException(
+      DependencyResolutionException resolutionException) throws EnforcerRuleException {
+    DependencyResolutionResult result = resolutionException.getResult();
+    DependencyNode root = result.getDependencyGraph();
+
+    for (Throwable cause = resolutionException.getCause();
+        cause != null;
+        cause = cause.getCause()) {
+      if (cause instanceof ArtifactTransferException) {
+        ArtifactTransferException artifactException = (ArtifactTransferException) cause;
+        Artifact artifact = artifactException.getArtifact();
+        String pathsToArtifact = findPaths(root, artifact);
+        ImmutableList<DependencyNode> firstArtifactPath = findFirstArtifactPath(root, artifact);
+        if (DependencyGraphBuilder.isUnacceptableResolutionExceptionAt(firstArtifactPath)) {
           logger.error("Could not find artifact " + artifact);
           logger.error("Paths to the missing artifact: " + pathsToArtifact);
-          break;
+          throw new EnforcerRuleException(
+              "Unable to build a dependency graph: " + resolutionException.getMessage(),
+              resolutionException);
+        } else {
+          logger.warn(
+              "There was missing artifact at "
+                  + pathsToArtifact
+                  + ". Continuing with partial dependency graph.");
         }
+        break;
       }
-      throw new EnforcerRuleException("Unable to build a dependency graph: " + e.getMessage(), e);
```

Comment:
This exception is moved to line 274 when the exception is invalidated by `DependencyGraphBuilder.isUnacceptableResolutionExceptionAt`